modules/integration/tests/src/test/java/org/wso2/appmanager/integration/test/cases/WebAppSubscribeTestCase.java

```diff
@@ -0,0 +1,96 @@
+/*
+*Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+*
+*WSO2 Inc. licenses this file to you under the Apache License,
+*Version 2.0 (the "License"); you may not use this file except
+*in compliance with the License.
+*You may obtain a copy of the License at
+*
+*http://www.apache.org/licenses/LICENSE-2.0
+*
+*Unless required by applicable law or agreed to in writing,
+*software distributed under the License is distributed on an
+*"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+*KIND, either express or implied.  See the License for the
+*specific language governing permissions and limitations
+*under the License.
+*/
+
+package org.wso2.appmanager.integration.test.cases;
+
+import org.json.JSONObject;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.appmanager.integration.utils.APPMPublisherRestClient;
+import org.wso2.appmanager.integration.utils.APPMStoreRestClient;
+import org.wso2.appmanager.integration.utils.AppmTestConstants;
+import org.wso2.appmanager.integration.utils.bean.SubscriptionRequest;
+import org.wso2.carbon.automation.engine.context.AutomationContext;
+import org.wso2.carbon.automation.engine.context.TestUserMode;
+import org.wso2.carbon.automation.engine.context.beans.User;
+import org.wso2.carbon.automation.test.utils.http.client.HttpResponse;
+
+public class WebAppSubscribeTestCase {
+
+    private static final String TEST_DESCRIPTION = "Verify Subscribing to a Web App";
+    private APPMPublisherRestClient appmPublisherRestClient;
+    private APPMStoreRestClient appmStoreRestClient;
+    private String appName = "WebAppSubscribeTestCase";
+    private String appVersion = "1.0.0";
+    private String context = "/" + appName;
+    private String trackingCode = "AM_" + appName;
+    private String appCreatedUser;
+
+
+    @BeforeClass(alwaysRun = true)
+    public void startUp() throws Exception {
+        AutomationContext appMServer = new AutomationContext(AppmTestConstants.APP_MANAGER,
+                                                             TestUserMode.SUPER_TENANT_ADMIN);
+        String backEndUrl = appMServer.getContextUrls().getWebAppURLHttps();
+        appmPublisherRestClient = new APPMPublisherRestClient(backEndUrl);
+        User appCreator = appMServer.getSuperTenant().getTenantUser("AdminUser");
+        appCreatedUser = appCreator.getUserName();
+        appmPublisherRestClient.login(appCreatedUser, appCreator.getPassword());
+        HttpResponse appCreateResponse = appmPublisherRestClient.webAppCreate(appName, context, appVersion, trackingCode);
+        JSONObject appCreateResponseData = new JSONObject(appCreateResponse.getData());
+        String uuid = appCreateResponseData.getString(AppmTestConstants.ID);
+        appmPublisherRestClient.publishWebApp(uuid);
+        User subscriber = appMServer.getSuperTenant().getTenantUser("Subscriber");
```

Comment:
Fixed in https://github.com/wso2/product-app-manager/commit/f021f86c3c398dddeeca17e225a37e42f7f7033b